Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -0,0 +1,711 @@
+  //  Clears _bocList_selectedRows array and sets _bocList_selectedRowsLength to zero.
+  //  bocList: The BocList whose rows should be unselected.
+  //  isRowHighlightingEnabled: true to update the row's css-class.
+  private static UnselectAllRows(bocList: HTMLElement, isRowHighlightingEnabled: boolean): void
+  {
+    var selectedRows = BocList._selectedRows[bocList.id]!;
+    for (var rowID in selectedRows.Rows)
+    {
+      var rowBlock = selectedRows.Rows[rowID];
+      if (rowBlock != null)
+      {
+        BocList.UnselectRow (bocList, rowBlock, isRowHighlightingEnabled);
+      }
+    }
+    
+    //  Start over with a new array
+    selectedRows.Clear();
+  }
+
+  //  Unselects a row.
+  //  Removes the row frin the _bocList_selectedRows array and decrements _bocList_selectedRowsLength.
+  //  bocList: The BocList to which the row belongs.
+  //  rowBlock: The row to be unselected.
+  //  isRowHighlightingEnabled: true to update the row's css-class.
+  private static UnselectRow(bocList: HTMLElement, rowBlock: BocList_RowBlock, isRowHighlightingEnabled: boolean): void
+  {
+    //  Remove currentRow from list
+    var selectedRows = BocList._selectedRows[bocList.id]!;
+    selectedRows.Rows[rowBlock.SelectorControl.id] = null;
+    selectedRows.Length--;
+
+    // Unselect currentRow
+    rowBlock.SelectorControl.checked = false;
+    if (isRowHighlightingEnabled)
+      $(rowBlock.Row).removeClass(BocList.TrClassNameSelected);
+
+    BocList.ClearSelectAllRowsSelector (selectedRows);
+  }
+
+  private static SetSelectAllRowsSelectorOnDemand (selectedRows: BocList_SelectedRows): void
+  {
+    if (selectedRows.DataRowCount == selectedRows.Length && selectedRows.DataRowCount > 0)
+      selectedRows.SelectAllSelectorControls!.each (function (this: JQuery) { this.checked = true; });
+  }
+
+  private static ClearSelectAllRowsSelector (selectedRows: BocList_SelectedRows): void
+  {
+    selectedRows!.SelectAllSelectorControls!.each (function (this: JQuery) { this.checked = false; });
+  }
+
+  //  Event handler for the selection selectorControl in the title row.
+  //  Applies the checked state of the title's selectorControl to all data rows' selectu=ion selectorControles.
+  //  bocList: The BocList to which the selectorControl belongs.
+  //  selectRowControlName: The name of the row selector controls.
+  //  isRowHighlightingEnabled: true to enable highting of the rows
+  public static OnSelectAllSelectorControlClick (bocList: HTMLElement, selectAllSelectorControl: JQuery, isRowHighlightingEnabled: boolean): void
+  {
+    var selectedRows = BocList._selectedRows[bocList.id]!;
+
+    if (selectedRows.Selection != BocList.rowSelectionMultiple)
+      return;
+    //  BocList_SelectRow will increment the length, therefor initialize it to zero.
+    if (selectAllSelectorControl.checked)
+      selectedRows.Length = 0;
+  
+    // TODO: Missing null check?
+    selectedRows.SelectRowSelectorControls!.each (function (this: JQuery)
```

Comment:
No, this can't be null at this point. but we probably want a follow-up task to make the BocList_SelectedRows object more than a DTO by moving all the select-features onto that type. Please add a ticket for this and probably should add a TODO reference where we use Bang with selectedRows.Select...

Reply:
@MichaelKetting Done, see [RM-7711](https://re-motion.atlassian.net/browse/RM-7711)